### ISSUE TITLE
Packages to provide dependencies are automatically selected.

### ISSUE THREAD

**Issue by zsinskri** [actionless/pikaur#51]
When a dependency is provided by multiple packages, the first provider is automatically selected.
Example:
```
pikaur -R rust rustup
pikaur -S firefox-kde-opensuse
[...]
:: There are 2 providers available for cargo:
:: Repository community           
   1) rust  2) rustup 
            
Enter a number (default=1):
warning: skipping target: rust                             
resolving dependencies...                          
looking for conflicting packages...        
                                   
Packages (1) rust-1:1.24.0-1
                   
Total Installed Size:  208.57 MiB
   
:: Proceed with installation? [Y/n]
```

`firefox-kde-opensuse` (make-)depends on `cargo` which is provided by `rust` and `rustup`. The user is never prompted, which package they want to install, the first one (`rust`) is always chosen. I have not yet tested this with non-make-deps or any other package than rust.

Expected behaviour:
Ask which package should be installed.

Versions:
Pikaur 0.7+28+g65d68b6
Pacman v5.0.2
ibalpm v10.0.2

**Comment by zsinskri** [comment]
I just realised `firefox-kde-opensuse` with `rust`/`rustup` was a really bad example as it also depends directly on `rust`.
But `arch-audit-git` shows the same behaviour and only depends on `cargo`, not `rust`.

**Comment by ixjlyons** [comment]
This happens because the `--noconfirm` flag is used in the command constructed [here](https://github.com/actionless/pikaur/blob/3127526a0cf6f9e5a1ca1344afe00cfb29f21e4f/pikaur/build.py#L327) when installing repo dependencies. You can demonstrate (assuming you don't have rust or rustup installed) by comparing:

```
$ sudo pacman -S --noconfirm cargo
... automatically starts downloading and installing the rust package
```

to

```
$ sudo pacman -S cargo
... prompts which package to install as expected
```

I'm guessing it's there to minimize user interactions/confirmations. Having an additional confirmation prompt for repo dependency installation (i.e. just removing the `--noconfirm` flag here) would be fine to me as a pikaur user. I think I would actually prefer that behavior.

As a workaround for now, you can do the following:

```
$ pikaur -S arch-audit-git --confirm
```

(Though in this specific case, you only get a bit further along before running into another issue which is that you need to install the rust nightly toolchain before building the arch-audit-git package if you choose rustup instead of rust)

**Comment by actionless** [comment]
that's actually related to another problem which i described here: https://wiki.archlinux.org/index.php/Talk:AUR_helpers#Add_.22pacman_wrap.22_column

tl;dr -- pacman doesn't have option like --noconfirm but to skip only first question

**Comment by ixjlyons** [comment]
Yeah, `--noconfirm` to me sounds like it simply answers "yes" to the question "Proceed with installation? [Y/n]" for you, allowing for further user interaction later on (like choosing from a list of alternative packages). It looks like it's more for script usage in situations where you never want any user interaction at all. If there are errors along the way, it just exits. I guess it makes sense if you're automating Arch setups.

My "vote" (if you're counting them) on this would be to, at least temporarily, remove the `--noconfirm` flag and deal with the extra confirm step when there are repo dependencies. You're already going through a series of confirmation steps at this point in usage of pikaur (want to install? want to edit pkgbuild?, etc.), and if people *really* do not want to interact with pikaur, they can add the `--noconfirm` flag themselves. This way, pikaur isn't seemingly making decisions on your behalf by default. I'm happy to submit a PR.

I suppose you could try to recognize this scenario (repo dep has multiple providers) and remove the `--noconfirm`, but I would still prefer to be made aware of the fact that a new repo package is being installed. That information isn't very apparent until after it has happened in some cases.

**Comment by actionless** [comment]
yup, i think i'll change that in a similar way for now (because i'll also change the order of steps to avoid big time gap between the questions)

**Comment by actionless** [comment]
@ixjlyons see the linked commit (on a separate branch)

**Comment by ixjlyons** [comment]
I haven't done thorough testing, but it seems to work for me. I think relying more on makepkg is a good direction to go in and removes some code/complexity, so :+1: for me. I can't seem to reproduce the failure on Travis though.

If you can get something like `--assume-yes` added to pacman, I'd still like to be able to see and confirm that installing an AUR package will pull in repo dependencies before it happens. So, for example, it would be cool to have a single prompt that shows the AUR package to be installed, any dependencies in the AUR, and any repo dependencies. Then I can confirm all of that and then proceed through PKGBUILD viewing etc. It seems like doing that would require some bigger changes, though, and I don't think you'd be able to rely on makepkg UI for repo dependency installation in that case.

**Comment by actionless** [comment]
i merged that branch into the mainline
